### ISSUE TITLE
Where are gauges? 

### ISSUE THREAD

**Issue by onemenny** [mikejihbe/metrics#36]
Why are gauges missing from the library?

**Comment by tolbertam** [comment]
It looks like they are missing, the way I mimic them is using counters, i.e. whenever I go to update the metric:

```
    connectionCounter.clear();
    connectionCounter.inc(connectionCount);
```

Although the benefit of gauges is that you can configure when the values are updated, i.e. either on request or cached.  I think it'd be a good idea to add support.

**Comment by mikejihbe** [comment]
You're right, they're missing. Would definitely accept a PR.

**Comment by tolbertam** [comment]
I'd be happy to give this a shot as I have a few metrics that are counters that should probably be gauges.  I'll follow the dropwizard metrics library and provide implementations to match those ([docs](http://metrics.dropwizard.io/3.1.0/manual/core/#gauges)), except JMX gauges of course 😉

**Comment by rafalsiwiec** [comment]
Hey @tolbertam, is this going to happen in the nearest future? :)
If not, then I'm keen to prepare the PR.

**Comment by rafalsiwiec** [comment]
@tolbertam with this PR #59, I would like to start a discussion: what else gauges do we want to have?

- `CachedGauge` - with configurable expiration period seems ok
- `RatioGauge` and `Derivative Gauges` - I don't think they're actually needed

Also, I think that some utility for projecting results of async operations would be useful... What do you think?

**Comment by tolbertam** [comment]
Thanks for getting a start on this @rafalsiwiec!  I think I agree with your sentiment.  `CachedGauges` are very useful and I have used them quite a bit with dropwizard metrics.  I've used the other two rarely, they have their use but I don't think they are needed right away.  I'd be happy with default and `CachedGauges` to get started and we can always add the other implementations later.

I'll take a look at your PR tonight, thanks!
